<a href="https://colab.research.google.com/github/Jullian29t/programacion1/blob/main/Quiz_3_26_de_marzo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Quiz 26 marzo
>Julian Alexis Usma Forero - 2261346

In [52]:
%%writefile pedirnotas.py
def perdir_nombre():
    return input("ingrese el nombre del estudiante: ")

def pedir_notas():
    n1 = float(input("nota 1: "))
    n2 = float(input("nota 2: "))
    n3 = float(input("nota 3: "))
    n4 = float(input("nota 4: "))
    return [n1, n2, n3]

Overwriting pedirnotas.py


In [91]:
%%writefile promedio.py
import statistics

def calcular_promedio(notas_list):
    return statistics.mean(notas_list)

def mediana(notas_list):
    return statistics.median(notas_list)

def moda(notas_list):
    return statistics.mode(notas_list)

Overwriting promedio.py


In [93]:
import pedirnotas
pedirnotas.perdir_nombre()

ingrese el nombre del estudiante: Julian Usma


'Julian Usma'

In [100]:
import importlib
importlib.reload(pedirnotas)
notas = pedirnotas.pedir_notas()

nota 1: 5
nota 2: 3
nota 3: 3
nota 4: 2


In [101]:
import importlib
importlib.reload(promedio)

mode_notas = promedio.moda(notas)
promedio_notas = promedio.calcular_promedio(notas)
mediana_notas = promedio.mediana(notas)
print(f"El promedio de las notas es: {promedio_notas}")
print(f"La mediana de las notas es: {mediana_notas}")
print(f"La moda de las notas es: {mode_notas}")

El promedio de las notas es: 3.6666666666666665
La mediana de las notas es: 3.0
La moda de las notas es: 3.0


In [107]:
if promedio_notas >= 3:
    print("El estudiante está aprobado")
else:
    print("El estudiante está reprobado")

El estudiante está aprobado


El análisis de las notas del estudiante muestra que el promedio es de 3.67, la mediana es 3.0 y la moda también es 3.0. Dado que para aprobar la minima nota es 3.0, y el promedio del estudiante 3.67 lo cual hace q el estudiante haya sido aprobado